crates/bevy_reflect/derive/src/derive_data.rs

```diff
@@ -536,6 +539,52 @@ impl<'a> ReflectStruct<'a> {
     pub fn where_clause_options(&self) -> WhereClauseOptions {
         WhereClauseOptions::new_with_fields(self.meta(), self.active_types().into_boxed_slice())
     }
+
+    /// Returns the `Reflect::reflect_clone` impl, if any, as a `TokenStream`.
+    pub fn get_clone_impl(&self) -> Option<proc_macro2::TokenStream> {
+        let bevy_reflect_path = self.meta().bevy_reflect_path();
+
+        if let container_clone @ Some(_) = self.meta().attrs().get_clone_impl(bevy_reflect_path) {
+            return container_clone;
+        }
+
+        let mut tokens = proc_macro2::TokenStream::new();
+
+        for field in self.fields() {
+            let member = ident_or_index(field.data.ident.as_ref(), field.declaration_index);
+
+            match &field.attrs.clone {
+                CloneBehavior::Default => {
+                    if field.attrs.ignore.is_ignored() {
+                        return None;
+                    }
+
+                    tokens.extend(quote! {
+                        #member: #bevy_reflect_path::Reflect::reflect_clone(&self.#member)?.take().ok()?,
+                    });
+                }
+                CloneBehavior::Trait => {
+                    tokens.extend(quote! {
+                        #member: #FQClone::clone(&self.#member),
+                    });
+                }
+                CloneBehavior::Func(clone_fn) => {
+                    tokens.extend(quote! {
+                        #member: #clone_fn(&self.#member),
+                    });
+                }
+            }
+        }
+
+        Some(quote! {
+            #[inline]
+            fn reflect_clone(&self) -> #FQOption<#FQBox<dyn #bevy_reflect_path::Reflect>> {
```

Comment:
Wouldn't it make more sense to return `Result` rather than `Option`?

The Clone could fail if the inner member clones doesn't exist or fails